python/Ganga/new_tests/GPI/GangaList/TestCopy.py

```diff
@@ -43,7 +36,7 @@ def testCopy(self):
         gl2 = copy.copy(gl)
         assert len(gl2) == len(gl), 'lists must be equal'
         assert gl2 is not gl, 'list must be copies'
-        #assert gl[0] is gl2[0], 'the references must be copied' ## rcurrie 09/12/15 Not sure this is valid given we have to cope with GangaObjects here which may not be copyable!
+        assert gl[0] is gl2[0], 'the references must be copied'
```

Comment:
In explicit terms I understand what you mean is that the following is true:
```
j=Job(); j.someArg = GangaList()
gl2 = copy.copy(j.someArg)
assert stripProxy(j.someArg[0])._getParent() is stripProxy(j)
assert stripProxy(j.someArg[0])._getParent() is stripProxy(gl2[0])._getParent()
assert stripProxy(gl2[0])._getParent() is stripProxy(j)
```

I don't mind what the expected behavior was/is/will-be I'm only bugged that when we were having problems with the parent of objects not being set correctly this was causing some levels of problems.

I'm just bugged by the following being allowed:
```
j=Job(); j.someArg = GangaList()
gl2 = copy.copy(j.someArg)
gl2[0].someActionToChangeItsValue()
assert j.someArg[0] == gl2[0]
```

Although this is close to expected python behavior so we should probably just live with that.

I'm not sure that the user does ever get a clone of an object through the Proxy layer (they do mostly when it matters but not from a ```copy.copy```) and that they get the proxy of each object to play with but the test here shows they're the same proxy hence are pointing to the same object.